libethereum/EthereumCapability.cpp

```diff
@@ -409,9 +409,15 @@ EthereumCapability::EthereumCapability(shared_ptr<p2p::CapabilityHostFace> _host
     m_urng = std::mt19937_64(seed());
 }
 
-EthereumCapability::~EthereumCapability()
+void EthereumCapability::onStarting()
 {
-    terminate();
+    m_backgroundWork = true;
```

Comment:
If you want something fancy
```suggestion
    m_backgroundWork.store(true, std::memory_order_relaxed);
```

Similarly for load.

Reply:
I'll leave it; not convinced yet that it gives enough improvement here to account for more confusing code

Reply:
I was curious.
https://godbolt.org/z/11A51F